src/Serilog.Sinks.AzureDataExplorer/Sinks/AzureDataExplorerSink.cs

```diff
@@ -56,52 +54,76 @@ public AzureDataExplorerSink(AzureDataExplorerSinkOptions options)
             m_formatProvider = options.FormatProvider;
             m_databaseName = options.DatabaseName;
             m_tableName = options.TableName;
-            m_mappingName = options.MappingName;
+            var mappingName = options.MappingName;
+            m_flushImmediately = options.FlushImmediately;
+            m_streamingIngestion = options.UseStreamingIngestion;
 
             m_ingestionMapping = new IngestionMapping();
-            if (!string.IsNullOrEmpty(m_mappingName))
+            if (!string.IsNullOrEmpty(mappingName))
             {
-                m_ingestionMapping.IngestionMappingReference = m_mappingName;
+                m_ingestionMapping.IngestionMappingReference = mappingName;
             }
             else if (options.ColumnsMapping?.Any() == true)
             {
                 m_ingestionMapping.IngestionMappings = options.ColumnsMapping.Select(m => new ColumnMapping { ColumnName = m.ColumnName, ColumnType = m.ColumnType, Properties = new Dictionary<string, string>(1) { { MappingConsts.Path, m.ValuePath } } }).ToList();
             }
             else
             {
-                m_ingestionMapping.IngestionMappings = s_defaultIngestionColumnMapping;
+                m_ingestionMapping.IngestionMappings = SDefaultIngestionColumnMapping;
             }
 
             var kcsb = options.GetKustoConnectionStringBuilder();
+            var engineKcsb = options.GetKustoEngineConnectionStringBuilder();
 
             if (options.UseStreamingIngestion)
             {
-                m_queuedIngestClient = KustoIngestFactory.CreateStreamingIngestClient(kcsb);
+                m_ingestClient = KustoIngestFactory.CreateManagedStreamingIngestClient(engineKcsb, kcsb);
             }
             else
             {
-                m_queuedIngestClient = KustoIngestFactory.CreateQueuedIngestClient(kcsb);
+                m_ingestClient = KustoIngestFactory.CreateQueuedIngestClient(kcsb);
             }
         }
 
         public async Task EmitBatchAsync(IEnumerable<LogEvent> batch)
         {
             using (var dataStream = CreateStreamFromLogEvents(batch))
             {
-                var result = await m_queuedIngestClient.IngestFromStreamAsync(
-                    dataStream,
-                    new KustoIngestionProperties
-                    {
-                        DatabaseName = m_databaseName,
-                        TableName = m_tableName,
-                        Format = DataSourceFormat.multijson,
-                        IngestionMapping = m_ingestionMapping
-                    },
-                    new StreamSourceOptions
-                    {
-                        LeaveOpen = false,
-                        CompressionType = DataSourceCompressionType.GZip
-                    }).ConfigureAwait(false);
+                if (!m_streamingIngestion)
+                {
+                    await m_ingestClient.IngestFromStreamAsync(
+                        dataStream,
```

Comment:
Is this stream disposed? More importantly check if the direct API can be used

Reply:
yes m_ingestClient is disposed via the default dispose() method.